cmd/util/ledger/migrations/staged_contracts_migration.go

```
@@ -148,11 +153,131 @@ func (m *StagedContractsMigration) InitMigration(
 	return nil
 }
 
+// collectAndRegisterStagedContractsFromPayloads scan through the payloads and collects the contracts
+// staged through the `MigrationContractStaging` contract.
+func (m *StagedContractsMigration) collectAndRegisterStagedContractsFromPayloads(allPayloads []*ledger.Payload) error {
+
+	// If the contracts are already passed as an input to the migration
+	// then no need to scan the storage.
+	// TODO: Maybe do this anyway?
+	if len(m.contractsByLocation) > 0 {
+		return nil
+	}
+
+	var stagingAccount string
+
+	switch m.chainID {
+	case flow.Testnet:
+		stagingAccount = "0x2ceae959ed1a7e7a"
+	case flow.Mainnet:
+		stagingAccount = "0x56100d46aa9b0212"
+	default:
+		// For other networks such as emulator etc. no need to scan for staged contracts.
+		return nil
```

Maybe return an error, or at least log an error/warning?